makeflow/src/dag_node.c

```diff
@@ -18,6 +18,16 @@ See the file COPYING for details.
 #include <stdlib.h>
 #include <unistd.h>
 
+int dag_node_comp(void *item, const void *arg)
```

Comment:
Why do you compare by nodeid? Simply saying d == e should be enough.

Unless we have to objects in memory with the same nodeid. If that is so, something went really wrong.